src/graph.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
+# -*- coding: utf -8 -*-
+
+from __future__ import unicode_literals, division
+
+
+class Graph(object):
+    """Defining class Graph."""
+    def __init__(self, iterable=None):
+        """Initiate an instance of class Graph."""
+        self._dict = {}
+        try:
+            for item in iterable:
+                self._dict.setdefault(str(item), [])
+        except TypeError:
+            if iterable is not None:
+                self._dict.setdefault(str(iterable), [])
+
+    def add_node(self, n):
+        """Add a node to graph"""
+        return self._dict.setdefault(str(n), [])
+
+    def add_edge(self, n1, n2):
+        """Add a edge from n1 to n2"""
+        new_node = self._dict.setdefault(str(n1), [])
+        self._dict.setdefault(str(n2), [])
+        if str(n2) not in new_node:
+            new_node.append(str(n2))
+        else:
+            raise ValueError('This edge already exists.')
+
+    def nodes(self):
+        """Show all nodes"""
+        return self._dict.keys()
+
+    def edges(self):
+        """Show all edges"""
+        list_key_value = self._dict.items()
+        list_edges = []
+        for pair in list_key_value:
+            for node in pair[1]:
+                list_edges.append((pair[0], node),)
+        return list_edges
+
+    def del_node(self, n):
+        """Delete a node from graph"""
+        if str(n) in self._dict:
+            del self._dict[str(n)]
+            for key in self._dict:
+                if str(n) in self._dict[key]:
+                    node_value = self._dict[key]
+                    node_value.remove(str(n))
+        else:
+            raise KeyError('No such node in the graph.')
```

Comment:
You should probably be more consistent between your different errors here. Why throw KeyError for failure to delete a Node but ValueError for failure to delete an edge?